ledger/acctupdates.go

```diff
@@ -1825,26 +1894,88 @@ func (au *accountUpdates) lookupWithRewards(rnd basics.Round, addr basics.Addres
 		// present in the on-disk DB.  As an optimization, we avoid creating
 		// a separate transaction here, and directly use a prepared SQL query
 		// against the database.
-		dbad, err = au.accountsq.lookup(addr)
+		if full {
+			dbad, err = lookupFull(au.dbs.Rdb, addr)
+			var mismatchRoundErr *MismatchingDatabaseRoundError
+			if errors.As(err, &mismatchRoundErr) {
+				goto retryAcquireLock
+			}
```

Comment:
MismatchingDatabaseRoundError might happen when round advances between accountbase and accountext reading, not sure to to test it.

Reply:
yes, I agree it might be a hard thing to test.